storage/innobase/handler/ha_innodb.cc

```diff
@@ -18429,6 +18446,14 @@ static void innodb_data_file_write_through_update(THD *, st_mysql_sys_var*,
   mysql_mutex_lock(&LOCK_global_system_variables);
 }
 
+static void innodb_doublewrite_update(THD *, st_mysql_sys_var*,
+                                      void *, const void *save)
+{
+  if (!fil_system.set_use_doublewrite(*static_cast<const ulong*>(save)))
+    my_error(ER_INCORRECT_GLOBAL_LOCAL_VAR, MYF(0),
+             "innodb_doublewrite=OFF", "read only");
+}
+
```

Comment:
9c4f333088b23dcb6c06b0c96dedf39028ab8a51 is a change of approach: changing the Boolean `innodb_doublewrite` to an `ENUM` of 5 values, instead of changing `innodb_data_file_write_through`.

I realized that we can probably lift this limitation and allow any `SET GLOBAL innodb_doublewrite` changes if we invoke `fil_space_t::reopen_all()` when the value of `buf_dblwr.in_use()` would change. I will implement that if @vaintroub is fine with the rest of these changes.

Reply:
I lifted this limitation by making `IORequest` remember whether the request was initiated via `buf_dblwr_t::add_to_batch()`.